identity/aziot-identityd/src/lib.rs

```diff
@@ -204,6 +204,83 @@ impl Api {
         Err(Error::Authorization)
     }
 
+    pub async fn get_provisioning_info(
+        &self,
+    ) -> Result<aziot_identity_common_http::get_provisioning_info::Response, Error> {
+        match &self.settings.provisioning.provisioning {
+            config::ProvisioningType::Dps {
+                global_endpoint,
+                scope_id,
+                attestation,
+            } => {
+                let (auth, registration_id) = match attestation {
+                    config::DpsAttestationMethod::SymmetricKey {
+                        registration_id, ..
+                    } => ("symmetric_key".to_string(), registration_id.to_string()),
+                    config::DpsAttestationMethod::Tpm { registration_id } => {
+                        ("tpm".to_string(), registration_id.to_string())
+                    }
+                    config::DpsAttestationMethod::X509 { identity_cert, .. } => {
+                        // The identity certificate's common name is used as registration ID and always
+                        // overrides any registration ID set in the config. Therefore, the registration
+                        // ID must always be retrieved from the identity certificate.
```

Comment:
`provision_device` prefers the `registration_id` in the config over the cert CN, no?

Reply:
So it's a bit complicated:
- If an identity certificate exists and is not within a day of expiry, IS uses that certificate's CN as registration ID and ignores the registration ID in the config.
- Otherwise, if no cert exists or cert is expired, the priority is:
  - config's registration ID
  - old certificate's CN

For this API, which is only available after a successful provision (i.e. when the device identity certificate has been successfully used and therefore must exist and be valid), the CN of the certificate will always be the registration ID.

Reply:
I updated this comment to make it clearer.